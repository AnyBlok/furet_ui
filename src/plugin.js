/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
class Plugin {
    constructor () {
        this.plugs = {};
    }
    set (paths=[], obj) {
        const recPlugs = (plugs, _paths) => {
            if (_paths.length == 0) {
                Object.assign(plugs, obj);
            } else {
                if (! plugs[_paths[0]]) plugs[_paths[0]] = {};
                recPlugs(plugs[_paths[0]], _paths.slice(1))
            }
        };
        recPlugs(this.plugs, paths)
    }
    get (paths=[]) {
        const recPlugs = (plugs, _paths) => {
            if (_paths.length == 1) {
                return plugs[_paths[0]];
            } else {
                return recPlugs(plugs[_paths[0]] || {}, _paths.slice(1))
            }
        };
        return recPlugs(this.plugs, paths)
    }
}

const plugin = new Plugin();
export default plugin;
