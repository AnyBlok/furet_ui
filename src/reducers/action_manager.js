/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
export const defaultState = {
    actions: [],
    action_data: {},
}

export const action_manager = (state = defaultState, action) => {
    const value = Object.assign({}, action);
    delete value.type;
    switch (action.type) {
        case 'UPDATE_ACTION_MANAGER_ADD_ACTION':
            const actions = state.actions.slice(0);
            actions.push(action.actionId);
            return Object.assign({}, state, {actions});
        case 'UPDATE_ACTION_MANAGER_ADD_ACTION_DATA':
            delete value.actionId;
            const action_data = Object.assign({}, state.action_data);
            action_data[action.actionId] = value;
            return Object.assign({}, state, {action_data});
        case 'RESET_ACTION_MANAGER':
            return Object.assign({}, state, {actions: []});
        case 'CLEAR_ACTION_MANAGER':
            return defaultState;
        default:
            return state
    }
}

export default action_manager;
