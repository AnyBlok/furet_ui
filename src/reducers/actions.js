const defaultState = {}

export const actions = (state = defaultState, action) => {
    const value = Object.assign({}, action);
    delete value.type;
    switch (action.type) {
        case 'UPDATE_ACTION_SELECT_VIEW':
            delete value.actionId;
            const values = {};
            values[action.actionId] = Object.assign({}, state[action.actionId], value);
            return Object.assign({}, state, values);
        case 'CLEAR_ACTION':
            return defaultState;
        default:
            return state
    }
}

export default actions;
