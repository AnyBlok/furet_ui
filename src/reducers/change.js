/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import _ from 'underscore';
export const defaultState = {
    current: {},
    toSync: [],
    computed: {},
    currents: {},
}

export const current2Sync = (current, toSync, uuid, model, dataId, newData, fields) => {
    const newSync = {
        uuid,
        state: 'toSend',
        data: [],
    }
    // Add main data in first
    try {
        newSync.data.push({
            model,
            dataId,
            type: newData ? 'CREATE' : 'UPDATE',
            fields,
            data: current[model][dataId],
        });
    } catch (e) {};
    // Add other data
    _.each(_.keys(current), modelName => {
        _.each(_.keys(current[modelName]), id => {
            if (modelName == model && dataId == id) {
                // do nothing, because already done
            } else {
                const data = current[modelName][id];
                if (data == 'DELETED'){
                    newSync.data.push({
                        model: modelName,
                        dataIds: [id],
                        type: 'DELETE',
                    });
                }else {
                    const _fields = current[modelName][id].__fields;
                    delete current[modelName][id].__fields;
                    newSync.data.push({
                        model: modelName,
                        dataId: id,
                        type: (new RegExp('^new-.*')).test(id) ? 'CREATE' : 'UPDATE',
                        fields: _fields,
                        data,
                    });
                }
            }
        });
    });
    if (newSync.data.length > 0) {
        toSync.push(newSync);
    }
}

export const remove2Sync = (toSync, uuid, model, dataIds) => {
    if (dataIds.length == 0) return;
    toSync.push({
        uuid,
        state: 'toSend',
        data: [
            {
                model,
                dataIds,
                type: 'DELETE',
            }
        ],
    });
}

export const toSync2computed = (toSync, computed) => {
    _.each(toSync, ts => {
        _.each(ts.data, data => {
            if (computed[data.model] == undefined) computed[data.model] = {};
            switch (data.type) {
                case 'CREATE':
                case 'UPDATE':
                    if (computed[data.model][data.dataId] == undefined) computed[data.model][data.dataId] = {};
                    Object.assign(computed[data.model][data.dataId], data.data);
                    break;
                case 'DELETE':
                    _.each(data.dataIds, dId => {
                        computed[data.model][dId] = 'DELETED';
                    });
                    break;
            }
        });
    });
}

export const removeUuid = (toSync, uuid) => {
    return _.filter(toSync, s => s.uuid != uuid);
}

export const changeState = (toSync, uuid, state) => {
    return _.map(toSync, s => {
        const val = Object.assign({}, s)
        if (uuid == s.uuid) {
            val.state = state;
        }
        return val;
    });
}

export const change = (state = defaultState, action) => {
    const current = Object.assign({}, state.current),
          toSync = state.toSync.slice(0),
          computed = {},
          currents = Object.assign({}, state.currents);
    switch (action.type) {
        case 'ON_CHANGE':
            if (current[action.model] == undefined) current[action.model] = {};
            else current[action.model] = Object.assign({}, current[action.model]);
            if (current[action.model][action.dataId] == undefined) current[action.model][action.dataId] = {};
            else current[action.model][action.dataId] = Object.assign({}, current[action.model][action.dataId]);
            current[action.model][action.dataId][action.fieldname] = action.newValue;
            if (action.fields) {
                current[action.model][action.dataId]['__fields'] = action.fields;
            }
            return Object.assign({}, state, {current});
        case 'ON_CHANGE_DELETE':
            if (current[action.model] == undefined) current[action.model] = {};
            else current[action.model] = Object.assign({}, current[action.model]);
            _.each(action.dataIds, dataId => {
                current[action.model][dataId] = 'DELETED'
            });
            return Object.assign({}, state, {current});
        case 'CLEAR_CHANGE':
            return Object.assign({}, state, {current: {}});
        case 'ON_SAVE':
            current2Sync(current, toSync, action.uuid, action.model, action.dataId, action.newData, action.fields);
            toSync2computed(toSync, computed)
            return Object.assign({}, state, {current: {}, toSync, computed});
        case 'ON_DELETE':
            remove2Sync(toSync, action.uuid, action.model, action.dataIds);
            toSync2computed(toSync, computed)
            return Object.assign({}, state, {current: {}, toSync, computed});
        case 'SYNC':
            return Object.assign({}, state, {toSync: changeState(toSync, action.uuid, 'Sent')});
        case 'UNSYNC':
            return Object.assign({}, state, {toSync: changeState(toSync, action.uuid, 'toSend')});
        case 'SYNCED':
            let _toSync = removeUuid(toSync, action.uuid);
            toSync2computed(_toSync, computed)
            return Object.assign({}, state, {toSync: _toSync, computed});
        case 'ADD_CURRENTS':
            const c = {};
            c[action.actionId] = current;
            Object.assign(currents, c);
            return Object.assign({}, state, {current: {}, currents});
        case 'REVERT_CHANGE':
            const prev_current = currents[action.actionId];
            delete currents[action.actionId];
            _.each(action.actionIds, actionId => {
                delete currents[actionId];
            });
            return Object.assign({}, state, {current: prev_current, currents});
        default:
            return state
    }
}

export default change;
