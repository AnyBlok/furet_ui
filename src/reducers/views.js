/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
export const defaultState = {}

export const views = (state = defaultState, view) => {
    const value = Object.assign({}, view);
    delete value.type;
    switch (view.type) {
        case 'UPDATE_VIEW':
            delete value.viewId;
            const values = {};
            values[view.viewId] = Object.assign({}, state[view.viewId], value);
            return Object.assign({}, state, values);
        case 'CLEAR_ACTION':
            return defaultState;
        default:
            return state
    }
}

export default views;
