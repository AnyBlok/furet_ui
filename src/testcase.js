import _ from 'underscore';

export const updateGlobal = () => {
    global.document = {
        createElement: () => {
            return {
                innerHTM: () => {},
            };
        },
        body: {
            appendChild: () => {},
        }
    };
    global.window = {
        addEventListener: () => {},
        attachEvent: () => {},
    }
    global.navigator = global.navigator || {};
    global.navigator.userAgent = 'all';
    global._ = _;
}

export default {
    updateGlobal,
}
