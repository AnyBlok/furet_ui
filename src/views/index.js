/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import './clients';
import './search';
import './list';
import './thumbnail';
import './form';
import Vue from 'vue';
import plugin from '../plugin';

/**
 * Unknown view, use if no view found
 *
 * props:
 *  @viewName: Name of the custom view which is not available
 *  @viewType: Name of the standard view which is not available
 *
**/
export const ViewUnknown = Vue.component('furet-ui-view-unknown', {
    props: ['viewName'],
    template: `
        <div v-bind:style="{marginBottom: '20px'}">
            <section class="section box">
                <h1 class="title">{{$t('views.unknown.title', {name: viewName})}}</h1>
                <h2 class="subtitle">
                    {{$t('views.unknown.message')}}
                </h2>
            </section>
        </div>`,
});

plugin.set(['views'], {Unknown: {vue: 'furet-ui-view-unknown'}})


export const ViewIconUnknown = Vue.component('furet-ui-view-icon-unknown', {
    template: '<i class="fa fa-exclamation-triangle"></i>',
})

plugin.set(['views', 'icon'], {Unknown: 'furet-ui-view-icon-unknown'})
