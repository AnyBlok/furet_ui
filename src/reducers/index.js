/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import _ from 'underscore';
import global from './global';
import client from './client';
import rightmenu from './right-menu';
import leftmenu from './left-menu';
import spaces from './spaces';
import {action_manager} from './action_manager';
import actions from './actions';
import views from './views';
import data from './data';
import translate from 'counterpart';
import * as moment from 'moment';
import 'moment';


export const dispatchAll = (dispatch, datas) => {
    _.each(datas, data => {
        if (data.type) {
            switch (data.type) {
                case 'UPDATE_LOCALES':
                    _.each(data.locales, d => {
                        translate.registerTranslations(d.locale, {furetUI: d.counterpart});
                    });
                    break
                case 'SET_LOCALE':
                    translate.setLocale(data.locale)
                    moment.locale(data.locale) 
                default:
                    dispatch(data);
            }
        }
    });
}

export default {
    global,
    client,
    rightmenu,
    leftmenu,
    spaces,
    action_manager,
    actions,
    views,
    data,
}
