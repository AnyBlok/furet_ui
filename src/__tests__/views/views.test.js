/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/

import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import chai from 'chai';
import {createStore, combineReducers} from 'redux';
import {Provider} from 'react-redux';
import reducers from '../../reducers';
import {updateGlobal} from '../testcase';
import {Unknown, getClientView, getViewIcon, getView} from '../../views';
import plugin from '../../plugin';

test('Render Unknown View with viewName', () => {
    const store = createStore(combineReducers(reducers));
    updateGlobal();
    const component = renderer.create(
        <Provider store={store}>
            <MuiThemeProvider>
                <Unknown viewName="Test"/>
            </MuiThemeProvider>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Render Unknown View with viewType', () => {
    const store = createStore(combineReducers(reducers));
    updateGlobal();
    const component = renderer.create(
        <Provider store={store}>
            <MuiThemeProvider>
                <Unknown viewType="Test"/>
            </MuiThemeProvider>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Unknown View is in the plugin', () => {
    const _Unknown = plugin.get(['views', 'Unknown']);
    chai.expect(_Unknown).to.equal(Unknown);
});

test('getClientView with unknown view', () => {
    const store = createStore(combineReducers(reducers));
    updateGlobal();
    const component = renderer.create(
        <Provider store={store}>
            <MuiThemeProvider>
                {getClientView('Unknown')}
            </MuiThemeProvider>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('getClientView with Login view', () => {
    const store = createStore(combineReducers(reducers));
    updateGlobal();
    const component = renderer.create(
        <Provider store={store}>
            <MuiThemeProvider>
                {getClientView('Login')}
            </MuiThemeProvider>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Unknown Icon for standard view is in the plugin', () => {
    const UnknownIcon = plugin.get(['views', 'icon', 'Unknown']);
    const store = createStore(combineReducers(reducers));
    updateGlobal();
    const component = renderer.create(
        <Provider store={store}>
            <MuiThemeProvider>
                <UnknownIcon />
            </MuiThemeProvider>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('getViewIcon with unknown view', () => {
    const store = createStore(combineReducers(reducers));
    updateGlobal();
    const component = renderer.create(
        <Provider store={store}>
            <MuiThemeProvider>
                {getViewIcon('Unknown', () => {})}
            </MuiThemeProvider>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('getViewIcon with List view', () => {
    const store = createStore(combineReducers(reducers));
    updateGlobal();
    let clicked = false;
    const component = renderer.create(
        <Provider store={store}>
            <MuiThemeProvider>
                {getViewIcon('List', () => {clicked=true;})}
            </MuiThemeProvider>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    chai.expect(clicked).to.equal(false);
    tree.props.onClick();
    chai.expect(clicked).to.equal(true);
});

test('getView with Unknown view', () => {
    const store = createStore(combineReducers(reducers));
    updateGlobal();
    let clicked = false;
    const component = renderer.create(
        <Provider store={store}>
            <MuiThemeProvider>
                {getView('Unknown', '1', {})}
            </MuiThemeProvider>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
