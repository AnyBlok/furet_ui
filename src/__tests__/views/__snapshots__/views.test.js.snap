// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Unknown View with viewName 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-sm-8 col-md-8 col-lg-9"
    />
    <div
      className="col-xs-12 col-sm-4 col-md-4 col-lg-3"
    />
  </div>
  <div
    className="container"
  >
    <h1>
      The wanted view "Test" is unknown
    </h1>
    <p>
      Report this message to the administrator
    </p>
  </div>
</div>
`;

exports[`Render Unknown View with viewType 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-sm-8 col-md-8 col-lg-9"
    />
    <div
      className="col-xs-12 col-sm-4 col-md-4 col-lg-3"
    />
  </div>
  <div
    className="container"
  >
    <h1>
      The wanted view "Test" is unknown
    </h1>
    <p>
      Report this message to the administrator
    </p>
  </div>
</div>
`;

exports[`Unknown Icon for standard view is in the plugin 1`] = `
<svg
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "MozUserSelect": "none",
      "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "WebkitUserSelect": "none",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "inline-block",
      "fill": "currentColor",
      "height": 24,
      "msUserSelect": "none",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "userSelect": "none",
      "width": 24,
    }
  }
  viewBox="0 0 24 24"
>
  <path
    d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
  />
</svg>
`;

exports[`getClientView with Login view 1`] = `
<div
  className="container"
>
  <div
    className="col-xs-10 col-xs-offset-1 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3 col-lg-6 col-lg-offset-3"
  >
    <div
      className={undefined}
      style={
        Object {
          "MozBoxSizing": "border-box",
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "marginTop": 100,
          "minWidth": "100%",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "MozBoxSizing": "border-box",
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "backgroundColor": "#00bcd4",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <span
              style={
                Object {
                  "MozUserSelect": "none",
                  "WebkitUserSelect": "none",
                  "color": "#ffffff",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "msUserSelect": "none",
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }
            >
              Login
            </span>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`getClientView with unknown view 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-sm-8 col-md-8 col-lg-9"
    />
    <div
      className="col-xs-12 col-sm-4 col-md-4 col-lg-3"
    />
  </div>
  <div
    className="container"
  >
    <h1>
      The wanted view "Unknown" is unknown
    </h1>
    <p>
      Report this message to the administrator
    </p>
  </div>
</div>
`;

exports[`getView with Unknown view 1`] = `
<div
  style={
    Object {
      "margin": 20,
    }
  }
>
  <div>
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-8 col-md-8 col-lg-9"
      />
      <div
        className="col-xs-12 col-sm-4 col-md-4 col-lg-3"
      />
    </div>
    <div
      className="container"
    >
      <h1>
        The wanted view "Unknown" is unknown
      </h1>
      <p>
        Report this message to the administrator
      </p>
    </div>
  </div>
</div>
`;

exports[`getViewIcon with List view 1`] = `
<button
  disabled={false}
  href={undefined}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onTouchStart={[Function]}
  onTouchTap={[Function]}
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
      "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "background": "none",
      "border": 10,
      "boxSizing": "border-box",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": 0,
      "fontWeight": "inherit",
      "height": 36,
      "margin": 0,
      "muiPrepared": true,
      "outline": "none",
      "overflow": "visible",
      "padding": 2,
      "position": "relative",
      "textDecoration": "none",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "verticalAlign": null,
      "width": 36,
      "zIndex": 1,
    }
  }
  tabIndex={0}
  type="button"
>
  <div
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  >
    <svg
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "WebkitUserSelect": "none",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": 24,
          "msUserSelect": "none",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": 24,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M4 14h4v-4H4v4zm0 5h4v-4H4v4zM4 9h4V5H4v4zm5 5h12v-4H9v4zm0 5h12v-4H9v4zM9 5v4h12V5H9z"
      />
    </svg>
    ;
  </div>
</button>
`;

exports[`getViewIcon with unknown view 1`] = `
<button
  disabled={false}
  href={undefined}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onTouchStart={[Function]}
  onTouchTap={[Function]}
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
      "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "background": "none",
      "border": 10,
      "boxSizing": "border-box",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": 0,
      "fontWeight": "inherit",
      "height": 36,
      "margin": 0,
      "muiPrepared": true,
      "outline": "none",
      "overflow": "visible",
      "padding": 2,
      "position": "relative",
      "textDecoration": "none",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "verticalAlign": null,
      "width": 36,
      "zIndex": 1,
    }
  }
  tabIndex={0}
  type="button"
>
  <div
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  >
    <svg
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "WebkitUserSelect": "none",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "currentColor",
          "height": 24,
          "msUserSelect": "none",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": 24,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
      />
    </svg>
    ;
  </div>
</button>
`;
